fastlane_bot/helpers/carbon_trade_splitter.py

```diff
@@ -0,0 +1,195 @@
+import json
+from typing import List, Dict
+
+from fastlane_bot.helpers import TradeInstruction
+
+
+class CarbonTradeSplitter:
+    NATIVE = "native"
+    WRAPPED = "wrapped"
+    NEITHER = "neither"
+
+    def __init__(self, ConfigObj):
+        self.ConfigObj = ConfigObj
+
+    def split_carbon_trades(
+        self, trade_instructions: List[TradeInstruction]
+    ) -> List[TradeInstruction]:
+        """Splits Carbon trades that cannot be aggregated into a single trade action.
+
+        This function split a single Carbon trade into multiple trades for cases that include a mix of tokens or different Carbon deployments.
+        For example NATIVE/WRAPPED -> TKN -> NATIVE -> TKN & WRAPPED -> TKN.
+
+        Args:
+            trade_instructions: The list of TradeInstruction objects.
+
+        Returns:
+            The processed list of TradeInstruction objects after splitting incompatible trades.
+
+        """
+        new_trade_list = []
+        for trade in trade_instructions:
+            if not self._is_carbon_trade(trade):
+                new_trade_list.append(trade)
+                continue
+
+            carbon_exchanges = self._process_carbon_trades(trade)
+            self._create_new_trades_from_carbon_exchanges(
+                carbon_exchanges, trade, new_trade_list
+            )
+
+        return new_trade_list
+
+    def _is_carbon_trade(self, trade: TradeInstruction) -> bool:
+        """Checks if a trade is on a Carbon deployment.
+
+        Args:
+            trade: a single TradeInstruction object.
+
+        Returns:
+            True if the trade is on a Carbon deployment. False otherwise.
+
+        """
+        return trade.exchange_name in self.ConfigObj.CARBON_V1_FORKS
+
+    def _get_real_tkn(self, token_address: str, token_type: str):
+        """Returns the correct token address for the trade.
+
+        This function returns the real token address for the pool. If the token isn't the native/wrapped gas token address, it just returns the token.
+        If the token is native/wrapped, it will use the token_type to return the correct address.
+
+        Args:
+            token_address: the token address
+            token_type: the self.NATIVE, self.WRAPPED, or SELF.NEITHER
+
+        Returns:
+            The correct token address for the pool.
+
+        """
+        if token_address not in [
+            self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS,
+            self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS,
+        ]:
+            return token_address
+        return (
+            self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS
+            if token_type == self.NATIVE
+            else self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS
+        )
```

Comment:
This is a baffling implementation IMO.

The `token_type` input argument can be one of three (NATIVE or WRAPPED or NEITHER), but here you are referring to it as if it was one of two (NATIVE or not NATIVE).

This function "feels" like it was designated to handle a variety of different scenarios, but either some of these scenarios aren't valid, or the function itself has been overly-complicated.

As it stands, this function takes 2 input arguments, each of which can attain one of 3 different values:
1. Input argument `token_address`, which can be either the address of the native token or the address of the wrapped token or neither
2. Input argument `token_type`, which can be either NATIVE or WRAPPED or NEITHER.

This leads to an extremely obfuscated implementation scheme.

For example, why would this function ever receive the address of the native token or the address of the wrapped token, AND type NEITHER?

Please rewrite the function.